src/preprocessing-service/preprocess.py

```diff
@@ -73,7 +73,24 @@ async def mask_logs(nw, loop, queue):
         for index, row in payload_data_df.iterrows():
             masked_logs.append(masker.mask(row["log"]))
         payload_data_df["masked_log"] = masked_logs
-        await nw.publish("preprocessed_logs", payload_data_df.to_json().encode())
+        if "filename" in payload_data_df.columns:
```

Comment:
payload can contain both control plane and application logs. this approach assumes it's one or the other